Objective

In this lab, I will use pandas to calculate summary statistics for a crime dataset, investigate suspicious values, distinguish impossible values from extreme values, and document my decisions.

Investigation Process
Calculate summary statistics.
Identify suspicious values.
Inspect the rows containing those values.
Document each verdict and proposed action.

In [8]:

import pandas as pd

df = pd.read_csv("usarrests.csv", index_col=0)

df.describe()

,Murder,Assault,UrbanPop
count,50.00000,49.000000,50.00000
mean,7.78800,182.183673,74.20000
std,4.35551,130.877435,73.40828
min,0.80000,45.000000,6.00000
25%,4.07500,109.000000,53.25000
50%,7.25000,159.000000,66.00000
75%,11.25000,249.000000,77.75000
max,17.40000,879.000000,570.00000


In [9]:
# 1. Check for impossible UrbanPop percentages (> 100)
print(df[df['UrbanPop'] > 100])

# 2. Check for extreme or corrupted Assault figures (e.g., above typical range)
print(df[df['Assault'] > 400])

# 3. Check for missing values
print(df[df.isna().any(axis=1)])

      Murder  Assault  UrbanPop
Iowa     2.2     56.0       570
                Murder  Assault  UrbanPop
South Carolina    14.4    879.0        48
         Murder  Assault  UrbanPop
Georgia    17.4      NaN        60


1. The UrbanPop in Iowa 570 is impossible. A population percentage cannot exceed 100%. The value 570 is likely a data entry typo where a decimal point was missed (e.g., 57.0%) or an extra digit was added. Action taken: Since the original intended percentage cannot be guessed with absolute certainty without external source verification, this specific cell value should be set to NaN (Missing) to preserve the state's other valid metrics (Murder and Assault) without skewing the population distributions.
2. Assault number 879 in South Carolina is merely extreme. Unlike a percentage bounded by 100, crime rates are totals or ratios per capita that can technically scale drastically higher during severe outliers. While 879 is significantly higher than the next highest value, it remains physically possible. Action taken: Retain the value in the dataset. Labeling an extreme real-world variable as "impossible" risks erasing critical analytical data regarding extreme crime clusters.
3. Assault Data in Georgia is missing. The descriptive count shows 49 entries for Assault instead of 50, pointing directly to a missing field. Action taken: Keep the row but document the missing baseline. When calculating total metrics for Assault, explicitly use methods that handle or skip NaN values natively (such as skipna=True) so it does not nullify automated aggregate calculations. 